packages/flutter/lib/src/rendering/sliver_grid.dart

```diff
@@ -447,7 +447,7 @@ class SliverGridDelegateWithMaxCrossAxisExtent extends SliverGridDelegate {
     this.crossAxisSpacing = 0.0,
     this.childAspectRatio = 1.0,
     this.mainAxisExtent,
-  }) : assert(maxCrossAxisExtent != null && maxCrossAxisExtent >= 0),
+  }) : assert(maxCrossAxisExtent != null && maxCrossAxisExtent > 0),
```

Comment:
We also need to update the docs above to match this.

Reply:
Hrm, looks like I did this in https://github.com/flutter/flutter/pull/60726 which was resolving a bunch of trivial bugs. Looks like https://github.com/flutter/flutter/issues/13861 was the motivation... but looking now 7 months later I am not sure it makes sense. Or at least I cannot remember. Let me check a few things to see if I can jog my memory. 😕

Reply:
Ah, that change was to assert the `crossAxisExtent` of the SliverConstraints provided to `performLayout`. This is a contradiction on `maxCrossAxisExtent`.
I did a bit of archaeology, and it looks like the constructor assert was added in https://github.com/flutter/flutter/pull/9945 which asserts it is >= 0
At that point the assert that `maxCrossAxisExtent > 0` was already in `_debugAssertIsValid`, originating all the way back in #7780
So I think this is a bug that has been around for quite a while! Nice find @chinmoy12c!